src/Illuminate/Database/Console/ShowModelCommand.php

```diff
@@ -42,253 +33,36 @@ class ShowModelCommand extends DatabaseInspectionCommand
                 {--database= : The database connection to use}
                 {--json : Output the model as JSON}';
 
-    /**
-     * The methods that can be called in a model to indicate a relation.
-     *
-     * @var array
-     */
-    protected $relationMethods = [
-        'hasMany',
-        'hasManyThrough',
-        'hasOneThrough',
-        'belongsToMany',
-        'hasOne',
-        'belongsTo',
-        'morphOne',
-        'morphTo',
-        'morphMany',
-        'morphToMany',
-        'morphedByMany',
-    ];
-
     /**
      * Execute the console command.
      *
      * @return int
      */
     public function handle()
     {
-        $class = $this->qualifyModel($this->argument('model'));
-
         try {
-            $model = $this->laravel->make($class);
-
-            $class = get_class($model);
+            $info = (new ModelInfoExtractor($this->laravel))->handle(
```

Comment:
Why'd you pass the app instance manually instead of injecting your class inside the `handle` method args?

```php
public function handle(MyClass $isNowSortedAndRequiresNoManualInitializationThxDI)
```

Reply:
Good call. I'm used to the framework not explicitly typing properties, but I agree that this is cleaner.